README.md
![Visitor Badge](https://visitor-badge.laobi.icu/badge?page_id=adgellida.ubunsys)

<p align="center">
  <a href="https://www.pling.com/p/1179908" target="_blank"><img src="https://www.linux-apps.com/images_sys/superstore/linux-apps-logo.png" width="250"></a><a href="https://www.pling.com/p/1179908" target="_blank"><img src="https://raw.githubusercontent.com/adgellida/resources/master/images/download-badge.png" width="250" alt="Gitlab"></a>
</p>

<a href="https://github.com/adgellida/ubunsys/blob/master/READMEesES.md" target="_blank"><img src="https://raw.githubusercontent.com/adgellida/data-protection-list/main/images/languages/spain.png" width="80" img align="right"></a>
<a href="https://github.com/adgellida/ubunsys" target="_blank"><img src="https://raw.githubusercontent.com/adgellida/data-protection-list/main/images/languages/united-kingdom.png" width="80" style="vertical-align:middle;margin:0px 50px" img align="right"></a>

<img src="https://raw.githubusercontent.com/adgellida/ubunsys/master/logo.png" width="80"> ubunsys
=============================================

<p align="center">
    <a href="https://www.gnu.org/licenses/gpl-2.0.en.html" target="_blank"><img src="https://img.shields.io/badge/license-GPLv2-blue.svg" alt="GNU GPL v2"></a>
    <a href="https://github.com/adgellida/ubunsys/stargazers" target="_blank"><img src="https://img.shields.io/github/stars/adgellida/ubunsys.svg" alt="stars"></a>
    <a href="https://github.com/adgellida/ubunsys/network" target="_blank"><img src="https://img.shields.io/github/forks/adgellida/ubunsys.svg" alt="forks"></a>
    <a href="https://github.com/adgellida/ubunsys/issues?q=is%3Aopen" target="_blank"><img src="https://img.shields.io/github/issues/adgellida/ubunsys.svg" alt="open issues"></a>
    <a href="https://github.com/adgellida/ubunsys/releases/latest" target="_blank"><img src="https://img.shields.io/github/downloads/adgellida/ubunsys/total.svg" alt="total downloads"></a>
    <a href="https://github.com/adgellida/ubunsys/releases/latest" target="_blank"><img src="https://img.shields.io/github/downloads/adgellida/ubunsys/v2020.08.25/total.svg" alt="total downloads latest release"></a>
    <a href="https://github.com/adgellida/ubunsys/releases/latest" target="_blank"><img src="https://img.shields.io/badge/latest release-v2020.08.25-blue.svg" alt="latest release"></a>
    <a href="https://github.com/adgellida/ubunsys/commits/master" target="_blank"><img src="https://img.shields.io/badge/maintained-active-green.svg" alt="maintained"></a>
    <a href="https://github.com/adgellida/ubunsys/releases"><img src="https://img.shields.io/badge/platform-Linux-lightgrey.svg" alt="Platforms"></a>
    <a href="https://github.com/adgellida/ubunsys/releases"><img src="https://img.shields.io/badge/status-alpha-orange.svg" alt="Status"></a>
    <a href="https://github.com/LewisVo/Awesome-Linux-Software/#utilities"><img src="https://cdn.rawgit.com/sindresorhus/awesome/d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg" alt="Awesome"/></a>
	<a href="https://www.codacy.com/app/adgellida/ubunsys?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=adgellida/ubunsys&amp;utm_campaign=Badge_Grade"><img src="https://api.codacy.com/project/badge/Grade/f33896b9ff894ab3a0862ada0835deaa"/></a>
</p>

<p align="center">
	<a href="https://t.me/ubunsys_app" target="_blank"><img src="https://raw.githubusercontent.com/adgellida/ubunsys/master/src/images/td_logo.png" width="25" alt="telegram group"></a>
	<a href="https://twitter.com/ubunsys_app" target="_blank"><img src="https://img.shields.io/twitter/follow/ubunsys_app.svg?style=social&label=Follow" alt="twitter"></a>
	<a href="https://gitter.im/adgellida/ubunsys" target="_blank"><img src="https://badges.gitter.im/adgellida/ubunsys.svg" alt="chat"></a>
	<a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=84KHR9VLK7XK8" target="_blank"><img src="https://img.shields.io/badge/Donate-PayPal-green.svg" alt="Donate to project"></a> <a href="https://www.paypal.me/adgellida" target="_blank"><img src="https://img.shields.io/badge/Donate-PayPal.me-green.svg" alt="Donate to developer"></a>
	<a href="https://www.bountysource.com/teams/ubunsys" target="_blank"><img src="https://img.shields.io/bountysource/team/ubunsys/activity.svg" alt="Donate with Bountysource"></a>
</p>

<p align="left">
	<a href="https://forthebadge.com/" target="_blank"><img src="https://raw.githubusercontent.com/adgellida/ubunsys/master/src/images/made-with-qt.png" alt="forthebadge"></a>
	<a href="https://www.qt.io" target="_blank"><img src="https://img.shields.io/badge/qt version-v5.9.5-green.svg" alt="qt version"></a>
	<a href="http://www.somsubhra.com/github-release-stats/?username=adgellida&repository=ubunsys" target="_blank"><img src="https://raw.githubusercontent.com/adgellida/ubunsys/master/src/images/github-release-stats.png" alt="github release stats"></a>
</p>

Gives the user a powerful control center to toggle on and off many system settings in Ubuntu, it is able to change system configurations, system updates, executable improvements, small system fixes, and more at the touch of a button.

It's theoretically compatible with all Ubuntu versions.

Due to lack of time paused development.

<img src="https://i.imgur.com/0kxR5Vj.gif" width="450" height="450" />

<img src="https://i.imgur.com/9jecSPE.gif" width="450" height="450" />

**There are currently 2 branches of development:**

* [master](https://github.com/adgellida/ubunsys/tree/master) - Includes the most stable and lastest version.
* [dev](https://github.com/adgellida/ubunsys/tree/dev) - This branch is deprecated, only for experiments.

**This is a very dangerous app, if you don't know what you are touching or how to fix it, don't do it. This software is for testing and knowledge purposes. It's unstable software.**

This is a GUI for these scripts [ubuntupackages](https://github.com/adgellida/ubuntupackages) and [ubuntuScripts](https://github.com/adgellida/ubuntuScripts).

Other Apps based on this:

* [winsys](https://github.com/adgellida/winsys)

Tutorial
=============================================

Push [here](https://github.com/adgellida/ubunsys/wiki/Tutorial_en) to go to the wiki.

More info
=============================================

Push [here](https://github.com/adgellida/ubunsys/wiki/More-info) to go to the wiki.

How can I help?
=============================================

You can create a new [issue](https://github.com/adgellida/ubunsys/issues/new) with an error, proposing new functionality, adding new packages, help on coding, etc...

Be careful
=============================================

Be careful! The initial versions have a lot of bugs and disabled functions for a reason.

If you want to help with development on this software, please contact me.

If you need help with compiling and packaging I have basic knowhow, feel free to ask me.

I'm not a senior programmer, nor junior. I program as a hobby. So I'm not following professional coding practices.

I know that my code is horrifying and very well implemented. With patience and time I'll go to cleaning and improving it.

Credits and thanks to
=============================================

* Icon based on [this](https://thenounproject.com/term/terminal/715962/). Author [Bernar Novalyi](https://thenounproject.com/bernar.novalyi)
* Theme got from [here](https://github.com/ColinDuquesnoy/QDarkStyleSheet). Author [ColinDuquesnoy](https://github.com/ColinDuquesnoy)
* SQLite library based on [this](https://github.com/katecpp/sql_with_qt). Author [katecpp](https://github.com/katecpp)

debian/changelog
ubunsys (2020.08.25~ubuntu20.04) focal; urgency=medium

  * Added minor changes and applied bugfixes.

 -- Antonio David Gellida Lavara <adgellida@gmail.com>  Mon, 25 Aug 2020 08:29:00 +0200


src/about.cpp
#include "mainwindow.h"
#include <QMessageBox>
#include <QPixmap>
#include <QMovie>
#include <QLabel>
#include <dbmanager.h>

///////////////////////ABOUT///////////////////////

void MainWindow::on_actionAbout_triggered()
{
    QMessageBox about;

    about.setTextFormat(Qt::RichText);   //this is what makes the links clickable
    about.setText(tr("<p align='left'><b>ubunsys</b></p></br>"
                  "<p align='left'>Configurable app referent to packages, updates and configurations</p></br>"
                  "<p align='left'>Contains bugs. Alpha status.</p></br>"
                  "<p align='left'>.</p></br>"

                  "<p align='left'><b>Credits to:</b></p></br>"

                  "<p align='left'>Art by "
                  "<a href='https://www.instagram.com/napafau/' style='color:pink' >napafau</a></p></br>"

                  "<p align='left'>Icon based on <a href='https://thenounproject.com/term/terminal/715962/' style='color:pink' >this</a> Author "
                  "<a href='https://thenounproject.com/bernar.novalyi' style='color:pink' >Bernar Novalyi</a></p></br>"

                  "<p align='left'>Theme got from <a href='https://thenounproject.com/term/terminal/715962/' style='color:pink' >here</a> Author "
                  "<a href='https://github.com/ColinDuquesnoy' style='color:pink' >ColinDuquesnoyi</a></p></br>"

                  "<p align='left'>SQLite library based on <a href='https://github.com/katecpp/sql_with_qt' style='color:pink' >this</a> Author "
                  "<a href='https://github.com/katecpp' style='color:pink' >katecpp</a></p></br>"

                  "<p align='left'>.</p></br>"

                  "<p align='left'><b>Sources:</b></p></br>"

                  "<p align='left'><a href='https://github.com/adgellida/ubunsys' style='color:red' >Github Repo </a></p></br>"

                  "<p align='left'><a href='https://gitlab.com/adgellida/ubunsys' style='color:red' >Gitlab Repo</a></p></br>"

                  "<p align='left'><a href='https://www.pling.com/p/1179908' style='color:red' >Linux-apps</a></p></br>"

                  "<p align='left'><a href='https://sourceforge.net/projects/ubunsys' style='color:red' >Sourceforge</a></p></br>"

                  "<p align='left'><a href='https://launchpad.net/~adgellida/+archive/ubuntu/ubunsys' style='color:red' >Launchpad PPA</a></p></br>"

                  "<p align='left'>.</p></br>"

                  "<p align='left'><a href='' style='color:red' ></a></p></br>"));

    static const QString path (QDir::homePath() + "/.ubunsys/configurations/config.db");
    DbManager db(path);

    QString status = db.getStatus("ubunsysInstalledVersion");

    about.setInformativeText(qPrintable("Copyright (c) 2020 all rights reserved\n\n\GPL v2\n\nAntonio David Gellida Lavara\n\nadgellida@gmail.com\n\n" + status));

    about.setWindowTitle(tr("About"));

    about.setStandardButtons(QMessageBox::Ok);
    //about.setIconPixmap(QPixmap(":/images/panda.png"));

    about.setIconPixmap(QPixmap(":/images/ubunsys+panda.png"));
    about.setWindowIcon(QIcon(":/images/ubunsys.png"));
    //about.setIcon(":/images/ubunsys.png");
    about.setDefaultButton(QMessageBox::Ok);

    about.show();
    about.exec();
}
